docs/Kubernetes.md

```diff
@@ -0,0 +1,97 @@
+  - [inCluster mode](#using-incluster-mode)
+  - [kubeconfig mode](#using-kubeconfig-mode)
+
+## Kubernetes usage
+Maestro uses kubernetes for orchestrating game room instances. It uses a unique **namespace** for each scheduler, and a unique **pod** for each game room instance.
+
+We use [client-go](https://github.com/kubernetes/client-go) for communicating with kubernetes. The [Runtime](internal/core/ports/runtime.go) port
+is the interface used for managing resources, you can find all of the features we are using for managing k8s resources in it.
+
+The diagram below shows how maestro components interact with kubernetes for managing resources.
+
+```mermaid
+flowchart BT
+classDef borderless stroke-width:0px
+classDef darkBlue fill:#00008B, color:#fff
+classDef brightBlue fill:#6082B6, color:#fff
+classDef gray fill:#62524F, color:#fff
+classDef gray2 fill:#4F625B, color:#fff
+
+
+subgraph maestroSystem[ ]
+    subgraph k8s[ ]
+        A3[Kubernetes]
+    end
+    class k8s,A3 brightBlue
+    class A3, borderless
+    
+    subgraph WORKER[ ]
+        A7[Operation Execution Worker<br/><br/>manage kubernetes resources by creating/deleting/updating pods abnd namespaces]
+    end
+    class WORKER,A7 brightBlue
+    class WORKER,A7 borderless
+    WORKER--Create namespace<br/>HTTPS-->k8s
+    WORKER--Delete namespace<br/>HTTPS-->k8s
+    WORKER--Create pod<br/>HTTPS-->k8s
+    WORKER--Delete pod<br/>HTTPS-->k8s
+    
+    subgraph RUNTIME_WATCHER[ ]
+        A8[Runtime Watcher <br/><br/> watch for change events in managed pods]
+    end
+    class RUNTIME_WATCHER,A8 brightBlue
+    class RUNTIME_WATCHER,A8 borderless
+    RUNTIME_WATCHER--List/Watch pods<br/>HTTPS-->k8s
+    
+
+    
+end
+
+
+click A3 "/csymapp/mermaid-c4-model/blob/master/AWAComponent.md" "AWA"
+
+```
+
+### Runtime watcher
+The runtime watcher component maintains a worker process for each scheduler that keeps watching and processing _change
+events_ in pods resources. For doing that, it uses a [pods informer](https://medium.com/codex/explore-client-go-informer-patterns-4415bb5f1fbd),
+binding handlers for **add**, **update** and **delete** events for all pods managed by it.
+
+This component is not responsible for updating/creating/deleting
+kubernetes resources, all it does is to watch for changes and update its game room instances internal representation using redis.
+
+### Operation execution worker
+The worker uses kubernetes for managing pods and namespaces. It executes several [operations](Operations.md) that, alongside other side effects, will need to create, update, and delete namespaces and pods.
+
+One important note regarding how maestro creates pods: each new requested game room instance will be assigned to a pseudo-random port to be used as **HostPort**.
+Maestro uses the scheduler **PortRange** to generate the pseudo-random port. Currently, maestro does not check for HostPort conflict while creating new rooms. The final address
```

Comment:
This text: "Currently, maestro does not check for HostPort conflict while creating new rooms."
What do you think about moving it to a new line somehow like this:
> **⚠ Currently, maestro does not check for HostPort conflict while creating new rooms**

I think this might be interesting based on two factors: 
1. It's easier to remove this line when this feature is implemented
2. This information is VERY important for understanding strange behaviors when troubleshooting